hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -396,8 +396,14 @@ case class TableJoin(left: TableIR, right: TableIR, joinType: String, joinKey: I
   require(joinKey >= 0)
   require(left.typ.keyType.zip(right.typ.keyType).exists { case (leftKey, rightKey) =>
     (leftKey.size >= joinKey) && (rightKey.size >= joinKey) &&
-      (leftKey.truncate(joinKey) isIsomorphicTo rightKey.truncate(joinKey))
+    (leftKey.truncate(joinKey) isIsomorphicTo rightKey.truncate(joinKey))
   })
+  require(left.typ.valueType.fieldNames.toSet
+    .intersect(right.typ.valueType.fieldNames.toSet)
+    .isEmpty)
```

Comment:
I think this should be using `leftRVDType` and `rightRVDType` below. Key fields that aren't part of the join key shouldn't collide either.

Reply:
that's a good point, yes